### PR TITLE
Restructure repository to incorporate plugins and connectors.

### PR DESCRIPTION
**Pull request by akon-dey**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,244 +1,58 @@
...
+	<artifactId>ground</artifactId>
+	<version>0.1-SNAPSHOT</version>
+
+	<name>Ground</name>
+	<url>http://www.ground-metadata.org</url>
+	<prerequisites>
+		<maven>2.2.1</maven>
+	</prerequisites>
+
+	<properties>
```
Comment (vsreekanti) [inline pom.xml]: These properties don't need to be in the parent POM.

File: pom.xml [hunk 2]
```diff
@@ -1,244 +1,58 @@
...
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>edu.berkeley</groupId>
+	<artifactId>ground</artifactId>
+	<version>0.1-SNAPSHOT</version>
+
+	<name>Ground</name>
+	<url>http://www.ground-metadata.org</url>
+	<prerequisites>
+		<maven>2.2.1</maven>
```
Comment (vsreekanti) [inline pom.xml]: Why are we specifying an old version? 3.3.9 is the latest version.

File: pom.xml [hunk 3]
```diff
@@ -1,244 +1,58 @@
...
+		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+		<!-- Dropwizard version -->
+		<dropwizard.version>1.0.0-rc3</dropwizard.version>
+
+		<!-- Cassandra version -->
+		<cassandra.version>3.0.0</cassandra.version>
+
+		<!-- Titan version -->
+		<titan.version>1.0.0</titan.version>
```
Comment (vsreekanti) [inline pom.xml]: These (Dropwizard, Cassandra, Titan) properties should be in the `ground-core` POM.

File: pom.xml [hunk 4]
```diff
@@ -1,244 +1,58 @@
...
+				<enabled>false</enabled>
+			</snapshots>
+		</repository>
+	</repositories>
+
+	<packaging>pom</packaging>
+	<modules>
+		<module>ground-core</module>
+		<module>ground-hive-plugin</module>
+  </modules>
```
Comment (vsreekanti) [inline pom.xml]: Fix indentation.

File: pom.xml [hunk 5]
```diff
@@ -1,244 +1,58 @@
...
-                    </archive>
-                </configuration>
-            </plugin>
-        </plugins>
-    </build>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>edu.berkeley</groupId>
+	<artifactId>ground</artifactId>
```
Comment (vsreekanti) [inline pom.xml]: This should be `ground-parent`.

File: ground-hive-plugin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>edu.berkeley</groupId>
+		<artifactId>ground</artifactId>
+		<version>0.1-SNAPSHOT</version>
+		<relativePath>../</relativePath>
+	</parent>
+	<artifactId>ground-hive-plugin</artifactId>
+  <name>Ground Hive Plugin</name>
+  <url>http://maven.apache.org</url>
```
Comment (vsreekanti) [inline ground-hive-plugin/pom.xml]: Why is the URL for the Ground-Hive plugin `maven.apache.org`?

File: ground-hive-plugin/pom.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    </dependency>
+    <!-- http://mvnrepository.com/artifact/org.apache.hive/hive-metastore -->
+	<dependency>
+		<groupId>org.apache.hive</groupId>
+		<artifactId>hive-metastore</artifactId>
+		<version>2.2.0-SNAPSHOT</version>
+		<scope>provided</scope>
+	</dependency>
+  </dependencies>
+</project>
```
Comment (vsreekanti) [inline ground-hive-plugin/pom.xml]: Please fix formatting and add spacing similar to `ground-core/pom.xml`.

File: ground-hive-plugin/pom.xml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+		<relativePath>../</relativePath>
+	</parent>
+	<artifactId>ground-hive-plugin</artifactId>
+  <name>Ground Hive Plugin</name>
+  <url>http://maven.apache.org</url>
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+  <dependencies>
+    <dependency>
```
Comment (vsreekanti) [inline ground-hive-plugin/pom.xml]: Test dependencies should be at the end of the dependency list.

**Comment by vsreekanti** [comment]
Formatting: 
- Package imports in `.java` files: First non-Java standard library libraries organized alphabetically, second Java libraries organized alphabetically. 
- 4 space indents. 
- Empty functions should either have a `// unimplemented` or `// TODO: implement by...`. 
- Use `this` before instance variables. 
- Appropriate spacing between functions. 
- No spaces between parens and arguments to method calls. e.g.,: `obj.method(arg1, arg2, arg3)`.

File: ground-hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java [hunk 9]
```diff
@@ -0,0 +1,815 @@
...
+import org.apache.hadoop.hive.metastore.api.Role;
+import org.apache.hadoop.hive.metastore.api.RolePrincipalGrant;
+import org.apache.hadoop.hive.metastore.api.SQLForeignKey;
+import org.apache.hadoop.hive.metastore.api.SQLPrimaryKey;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.TableMeta;
+import org.apache.hadoop.hive.metastore.api.Type;
+import org.apache.hadoop.hive.metastore.api.UnknownDBException;
+import org.apache.hadoop.hive.metastore.api.UnknownPartitionException;
+import org.apache.hadoop.hive.metastore.api.UnknownTableException;
```
Comment (vsreekanti) [inline ground-hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java]: We should be able to collapse this into `import org.apache.hive.metastore.api.*`?

**Comment by vsreekanti** [comment]
Since we're merging this things in the vein of having good organization for future plugins, I think we might want it to look something like:

```
| ground
|--\ ground-core
|----\ src/
|--\ plugins
|----\ hive-plugin
|------\ src/
|----\ other plugin
...
```

File: pom.xml (hunk 3 above)
Comment (akon-dey) [inline pom.xml]: I want this to be inherited by all children to prevent version mismatches.

File: pom.xml (hunk 5 above)
Comment (akon-dey) [inline pom.xml]: The top-level artefact is called `ground` in its `pom.xml`. I could change that to `ground-parent` but I think that is confusing.

File: pom.xml (hunk 3 above)
Comment (vsreekanti) [inline pom.xml]: Understood. Sounds good.

**Comment by vsreekanti** [comment]
@akon-dey I'm going to be a stickler for consistency's sake, but there are two more things I'd like you to change: 
1. Please use _4_ space indents, not 8 space indents. 
2. Please expand tabs to spaces instead of having tab characters. 

Other than that, everything looks good. Once you make those changes, I'll merge.

**Comment by akon-dey** [comment]
The code should all be properly formatted now. Any newly created code will adhere to the 4 space indentation rule if the eclipse style file is used.

**Pull request merged**